solutions/fully-configurable/variables.tf

```diff
@@ -0,0 +1,356 @@
+  default     = {}
+
+  validation {
+    condition     = alltrue([for name, role in var.service_credential_names : contains(["Manager", "Writer", "Reader", "Event Source Manager", "Channel Editor", "Event Notification Publisher", "Status Reporter", "Device Manager", "Email Sender", "Custom Email Status Reporter"], role)])
+    error_message = "The specified service credential role is not valid. The following values are valid for service credential roles: 'Manager', 'Writer', 'Reader', 'Event Source Manager', 'Channel Editor', 'Event Notification Publisher', 'Status Reporter', 'Device Manager', 'Email Sender', 'Custom Email Status Reporter'"
+  }
+}
+
+variable "event_notifications_name" {
+  type        = string
+  description = "The name of the Event Notifications instance that is created by this solution. If a `prefix` input variable is specified, it is added to this name in the `<prefix>-value` format."
+  default     = "base-event-notifications"
+}
+
+variable "service_plan" {
+  type        = string
+  description = "The pricing plan of the Event Notifications instance. Possible values: `Lite`, `Standard`"
+  default     = "standard"
+  validation {
+    condition     = contains(["lite", "standard"], var.service_plan)
+    error_message = "The specified pricing plan is not available. The following plans are supported: `Lite`, `Standard`"
+  }
+}
+
+variable "service_endpoints" {
+  type        = string
+  description = "Specify whether you want to enable public, or both public and private service endpoints. Possible values: `public`, `public-and-private`"
+  default     = "public-and-private"
+  validation {
+    condition     = contains(["public", "public-and-private"], var.service_endpoints)
+    error_message = "The specified service endpoint is not supported. The following endpoint options are supported: `public`, `public-and-private`"
+  }
+}
+
+variable "event_notifications_tags" {
+  type        = list(string)
+  description = "The list of tags to add to the Event Notifications instance."
+  default     = []
+}
+
+variable "existing_event_notifications_instance_crn" {
+  type        = string
+  description = "The CRN of existing event notification instance. If not supplied, a new instance is created."
+  default     = null
+}
+
+########################################################################################################################
+# KMS
+########################################################################################################################
+
+variable "key_management_service_encryption_enabled" {
+  type        = bool
+  description = "Set to true to enable encryption on Event Notifications instance and Cloud Object Storage bucket."
```

Comment:
make sure to state KMS encryption. Otherwise it seems of set to false no encryption is used, and thats not the case.